src/main/java/de/hysky/skyblocker/skyblock/dungeon/GoldorWaypointsManager.java

```diff
@@ -0,0 +1,247 @@
+            if (PHASE_COMPLETE.matcher(message).matches()) {
+                currentPhase++;
+            } else {
+                String playerName;
+
+                if ((playerName = getPlayerName(TERMINAL_ACTIVATED.matcher(message))) != null) {
+                    removeNearestWaypoint(TERMINALS, playerName);
+                } else if ((playerName = getPlayerName(DEVICE_ACTIVATED.matcher(message))) != null) {
+                    removeNearestWaypoint(DEVICES, playerName);
+                } else if ((playerName = getPlayerName(LEVER_ACTIVATED.matcher(message))) != null) {
+                    removeNearestWaypoint(LEVERS, playerName);
+                } else if (message.equals(CORE_ENTRANCE)) {
+                    active = false;
+                }
+            }
+        } else {
+            if (message.equals(TERMINALS_START)) {
+                enableAll(TERMINALS);
+                enableAll(DEVICES);
+                enableAll(LEVERS);
+                active = true;
+            }
+        }
+    }
+
+    private static void renderWaypoints(WorldRenderContext context, ObjectArrayList<GoldorWaypoint> waypoints) {
+        for (GoldorWaypoint waypoint : waypoints) {
+            if (waypoint.phase == currentPhase && waypoint.shouldRender()) {
+                waypoint.render(context);
+            }
+        }
+    }
+
+    private static void render(WorldRenderContext context) {
+        if (active) {
+            renderWaypoints(context, TERMINALS);
+            renderWaypoints(context, DEVICES);
+            renderWaypoints(context, LEVERS);
+        }
+    }
+
+    private static class GoldorWaypoint extends NamedWaypoint {
+        public static final Codec<GoldorWaypoint> CODEC = RecordCodecBuilder.create(i -> i.group(
+                WaypointTargetKind.CODEC.fieldOf("kind").forGetter(w -> w.kind),
+                Codec.INT.fieldOf("phase").forGetter(customWaypoint -> customWaypoint.phase),
+                TextCodecs.CODEC.fieldOf("name").forGetter(NamedWaypoint::getName),
+                BlockPos.CODEC.fieldOf("pos").forGetter(customWaypoint -> customWaypoint.pos)
+        ).apply(i, GoldorWaypoint::new));
+
+        private static final Supplier<Type> TYPE_SUPPLIER = () -> SkyblockerConfigManager.get().dungeons.goldor.waypointType;
+
+        final WaypointTargetKind kind;
+        final int phase;
+
+        GoldorWaypoint(WaypointTargetKind kind, int phase, Text name, BlockPos pos) {
+            super(pos, name, TYPE_SUPPLIER, kind.colorComponents, 0.25F, true);
+            this.kind = kind;
+            this.phase = phase;
+        }
+
+        /**
+         * The different classes of waypoints
+         */
+        enum WaypointTargetKind implements StringIdentifiable {
+            TERMINAL(0, 255, 0),
+            DEVICE(0, 0, 255),
+            LEVER(255, 255, 0);
+
+            private static final Codec<WaypointTargetKind> CODEC = StringIdentifiable.createCodec(WaypointTargetKind::values);
```

Comment:
I would swap to using the `createBasicCodec` method since the one you're using returns a deprecated type pretty sure

Reply:
Huh wow, I'm surprised IDEA didn't trigger any warnings on this like it usually does, probably because it's not the method itself that's deprecated but return type.

Most of this code is heavily inspired by the existing dungeon waypoints code (hence the logger being mistakenly copied from it, oops), and speaking of which, `StringIdentifiable::createCodec` is also used there https://github.com/SkyblockerMod/Skyblocker/blob/master/src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/SecretWaypoint.java#L133

Reply:
Probably wasn't noticed then, I'm not sure why `EnumCodec` is deprecated as Mojang obviously doesn't provide JavaDocs. Either its to discourage use of it or they don't want you referencing the type itself.